### PR TITLE
Added command to dump SQL from schema

### PR DESCRIPTION
**Pull request by adamwojs**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | N/A
| **Type**                                   | feature
| **Target Ibexa version** | `v4.6`
| **BC breaks**                          | yes/no

This command is useful for generating upgrade script for new packages 

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [ ] Provided automated test coverage.
- [x] Checked that target branch is set correctly (`main` for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [ ] Asked for a review (ping `@ibexa/engineering`).

### CODE REVIEW

File: src/bundle/Command/DumpSqlCommand.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+use Ibexa\DoctrineSchema\Builder\SchemaBuilder;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+final class DumpSqlCommand extends Command
+{
+    protected static $defaultName = 'ibexa:schema:dump-sql';
```
Comment (alongosz) [inline src/bundle/Command/DumpSqlCommand.php]: `schema` is too generic IMO. Can refer also to GraphQL and maybe a few more.

- `ibexa:doctrine:schema:dump-sql`
- `ibexa:database:schema:dump-sql`
- `ibexa:db:schema:dump-sql`
- ... ?

File: src/bundle/Resources/config/services.yaml [hunk 2]
```diff
@@ -28,3 +28,7 @@ services:
             $defaultTableOptions: '%ibexa.schema.default_table_options%'
 
     Doctrine\Common\EventManager: ~
+
+    Ibexa\Bundle\DoctrineSchema\Command\DumpSqlCommand:
+        tags:
+            - { name: console.command }
```
Comment (alongosz) [inline src/bundle/Resources/config/services.yaml]: This shouldn't be necessary due to auto-configuration

File: src/bundle/Command/DumpSqlCommand.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if ($input->hasArgument('file')) {
+            $schema = $this->schemaBuilder->importSchemaFromFile($input->getArgument('file'));
+        } else {
+            $schema = $this->schemaBuilder->buildSchema();
+        }
```
Comment (alongosz) [inline src/bundle/Command/DumpSqlCommand.php]: If I don't give arguments, this block fails with:
```
  Argument 1 passed to Ibexa\DoctrineSchema\Builder\SchemaBuilder::importSchemaFromFile() must be of the type string, null given, called in vendor/ibexa/doctrine-schema/src/bundle/Command/DumpSqlCommand.php on line 46 
  ```
Seems `hasArgument` doesn't do what we want it to do... It's always present with default value, which is `null`.

```php
if (null !== $input->getArgument('file')) {
```
seems to be working, but maybe I'm missing something?

Side note: maybe a list of files instead? If we want to dump schema from multiple schema files.

File: src/bundle/Command/DumpSqlCommand.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        $io = new SymfonyStyle($input, $output);
+        foreach ($schema->toSql($this->db->getDatabasePlatform()) as $sql) {
```
Comment (alongosz) [inline src/bundle/Command/DumpSqlCommand.php:53]: Maybe not now, but as a 2nd iteration over this we could introduce `--platform` argument which builds db platform instance based on the needs. AFAICS there's no coupling between the actual connection and instances of `AbstractPlatform`. Connection is used to detect platform to use for the current connection.

// Now internal question you've asked makes more sense

**Review by alongosz — CHANGES_REQUESTED** [review]

File: src/bundle/Command/DumpSqlCommand.php (hunk 4 above)
Comment (adamwojs) [reply src/bundle/Command/DumpSqlCommand.php:53]: Yes, is definitely a great idea.

**Comment by adamwojs** [comment]
Thank you @alongosz for a great code review. I've applied your suggestions in https://github.com/ibexa/doctrine-schema/pull/17/commits/ff1e8dbaddb1f7d084ff339fbb82bd56c0e75d4d

**Review by alongosz — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by mikadamczyk — APPROVED** [review]

**Comment by Steveb-p** [comment]
A small issue with it is it cannot perform comparison against current schema, which would be a lot more valuable for updates. Dumping an existing schema via IDE is something that is simple on it's own. Modifications of schema are a lot more difficult.

**Review by Steveb-p — APPROVED** [review]

**Pull request merged**